docs/user-guides/plugin/development.md

```diff
@@ -0,0 +1,133 @@
+```
+
+After unpacking the `*.tgz` file there is a folder named `nginx` containing the Helm Chart.
+
+## Generating a PluginDefinition from a Helm Chart
+
+Using the files of the Helm Chart we will create a new Greenhouse _PluginDefinition_ using the `greenhousectl` CLI.
+
+```bash
+greenhousectl plugin generate ./nginx ./nginx-plugin
+```
+
+This will create a new folder `nginx-plugin` containing the _PluginDefinition_ in a nested structure.
+
+## Modifying the PluginDefinition
+
+The generated _PluginDefinition_ contains a `plugin.yaml` file which defines the _PluginDefinition_. But there are still a few steps required to make it work.
+
+### Specify the Helm Chart repository
+
+After generating the _PluginDefinition_ the `.spec.helmChart.repository` field in the `pluginDefinition.yaml` contains a TODO comment. This field should be set to the repository where the Helm Chart is stored.
+For the bitnami nginx Helm Chart this would be `oci://registry-1.docker.io/bitnamicharts`.
+
+### Specify the UI application
+
+A _PluginDefinition_ may specify a UI application that will be integrated into the Greenhouse UI. This tutorial does not cover how to create a UI application. Therefore the section `.spec.uiApplication` in the `pluginDefinition.yaml` should be removed.
+
+### Modify the Options
+
+The _PluginDefinition_ contains a section `.spec.options` which defines options that can be set when deploying the _Plugin_ to a Cluster. These options have been generated based on the Helm Chart values.yaml file. You can modify the options to fit your needs.
+
+In general the options are defined as follows:
+
+```yaml
+    options:
+        - default: true
+          value: abcd123
+          description: automountServiceAccountToken
+          name: automountServiceAccountToken
+          required: false
+          type: ""
+```
+
+_default_ specifies if the option should provide a default value. If this is set to true, the value specified will be used as the default value. The _Plugin_ can still provide a different value for this option.
+_description_ provides a description for the option.
+_name_ specifies the Helm Chart value name, as it is used within the Chart's template files.
+_required_ specifies if the option is required. This will be used by the Greenhouse Controllers to determine if a _Plugin_ is valid.
+_type_ specifies the type of the option. This can be any of `[string, secret, bool, int, list, map]`. This will be used by the Greenhouse Controllers to validate the provided value.
+
+For this tutorial we will remove all options.
+
+## Deploying a Plugin to the Kind Cluster
+
+After modifying the _PluginDefinition_ we can deploy it to the local Greenhouse cluster and create a _Plugin_ that will deploy the `nginx` to the onboarded cluster.
+
+```bash
+  kubectl --kubeconfig=./envtest/kubeconfig apply -f ./nginx-plugin/nginx/17.3.2/pluginDefinition.yaml
+  plugindefinition.greenhouse.sap/nginx-17.3.2 created
```

Comment:
```suggestion
 
```

Reply:
Not sure if you meant to print the result here?

Reply:
Yes this is intentional